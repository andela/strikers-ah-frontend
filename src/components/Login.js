/* eslint-disable require-jsdoc */
import React, { PureComponent } from 'react';

class Login extends PureComponent {
  render() {
    return (
      // eslint-disable-next-line react/jsx-filename-extension
      <div>
        <h1>login component</h1>
      </div>
    );
  }
}
export default Login;
