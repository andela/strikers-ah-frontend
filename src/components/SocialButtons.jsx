import React, { PureComponent } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faFacebook,
  faGooglePlus,
  faTwitter,
  faGithub,
} from '@fortawesome/free-brands-svg-icons';

/**
 * Social login component
 */
class SocialLogin extends PureComponent {
  /**
   * @author frank harerimana
   * @param {*} party
   * @returns {*} social action
   */
  handlerClick(party) {
    this.action = party;
    return `http://localhost:5000/api/auth/${this.action}`;
  }

  /**
   * @author frank harerimana
   * @returns {*} component
   */
  render() {
    return (
      <div className="log-btn">
        <div className="login-button-section">
          <div className="social-login-btn">
            <a className="social-btn" href={`${this.handlerClick('github')}`}>
              <FontAwesomeIcon className="fa-github fa-2x" icon={faGithub} />
            </a>
          </div>
          <div className="social-login-btn">
            <a className="social-btn" href={`${this.handlerClick('twitter')}`}>
              <FontAwesomeIcon className="fa-twitter fa-2x" icon={faTwitter} />
            </a>
          </div>
          <div className="social-login-btn">
            <a className="social-btn" href={`${this.handlerClick('facebook')}`}>
              <FontAwesomeIcon
                className="fa-facebook-f fa-2x"
                icon={faFacebook}
              />
            </a>
          </div>
          <div className="social-login-btn">
            <a className="social-btn" href={`${this.handlerClick('google')}`}>
              <FontAwesomeIcon
                className="fa-google-plus-g fa-2x"
                icon={faGooglePlus}
              />
            </a>
          </div>
        </div>
      </div>
    );
  }
}
export default SocialLogin;
