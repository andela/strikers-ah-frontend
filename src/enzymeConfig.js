import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-localstorage-mock';

export default Enzyme.configure({
  adapter: new Adapter(),
});
