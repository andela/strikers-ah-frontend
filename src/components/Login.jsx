import React, { PureComponent } from 'react';
import SocialButtons from './SocialButtons';
import '../styles/css/login.css';
import logo from '../styles/img/logo.png';

/**
 * @author frank harerimana
 */
class Login extends PureComponent {
  /**
   * @author frank harerimana
   * @returns {*} render component
   */
  render() {
    return (
      <div className="container">
        <div className="login-section">
          <div className="logo">
            <img src={logo} alt="authors haven" />
          </div>
          <div className="log-text">
            <form>
              <input type="text" placeholder="Email or Username" />
              <input type="password" placeholder="Password" />
              <a href="false" className="fo-btn">
                Forgot password
              </a>
              <button className="login-btn" type="submit">
                Login
              </button>
            </form>
          </div>
          <SocialButtons />
        </div>
      </div>
    );
  }
}
export default Login;
