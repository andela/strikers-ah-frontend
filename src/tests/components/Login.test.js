/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { mount } from 'enzyme';
import '../../enzymeConfig';
import Login from '../../components/Login';
import SocialButtons from '../../components/SocialButtons';

describe('`login.jsx`', () => {
  test('should render', () => {
    const wrapper = mount(<Login />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should render', () => {
    const wrapper = mount(<SocialButtons />);
    expect(wrapper).toMatchSnapshot();
  });
});
