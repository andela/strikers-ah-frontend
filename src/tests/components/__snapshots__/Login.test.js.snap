// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`login.jsx\` should render 1`] = `ReactWrapper {}`;

exports[`\`login.jsx\` should render 2`] = `ReactWrapper {}`;
